src/hooks/Guilds/useClickOutside.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect } from 'react';
+
+export const useClickOutside = (ref, callback) => {

```

Comment:
you should add the comment here again on the purpose of this hook. I don't immediately know what `useClickOutside` is supposed to do. Or you can make the name clearer